contracts/service-registry/src/client.rs

```diff
@@ -0,0 +1,169 @@
+use error_stack::ResultExt;
+use router_api::ChainName;
+
+use crate::msg::{ExecuteMsg, QueryMsg};
+use crate::{Service, Verifier, WeightedVerifier};
+
+type Result<T> = error_stack::Result<T, Error>;
+
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("failed query the service registry for active verifiers for service {service_name} and chain {chain_name}")]
+    QueryActiveVerifiers {
+        service_name: String,
+        chain_name: ChainName,
+    },
+
+    #[error("failed to query service registry for service {0}")]
+    QueryService(String),
+
+    #[error("failed to query service registry for verifier {verifier} of service {service_name}")]
+    QueryVerifier {
+        service_name: String,
+        verifier: String,
+    },
+}
+
+impl<'a> From<client::Client<'a, ExecuteMsg, QueryMsg>> for Client<'a> {
+    fn from(client: client::Client<'a, ExecuteMsg, QueryMsg>) -> Self {
+        Client { client }
+    }
+}
+
+pub struct Client<'a> {
+    client: client::Client<'a, ExecuteMsg, QueryMsg>,
+}
+
+impl<'a> Client<'a> {
+    // TODO: add execute methods
+
+    pub fn active_verifiers(
+        &self,
+        service_name: String,
+        chain_name: ChainName,
+    ) -> Result<Vec<WeightedVerifier>> {
+        self.client
+            .query(&QueryMsg::ActiveVerifiers {
+                service_name: service_name.clone(),
+                chain_name: chain_name.clone(),
+            })
+            .change_context_lazy(|| Error::QueryActiveVerifiers {
+                service_name,
+                chain_name,
+            })
```

Comment:
not sure lazy is really helpful in these cases. You still need to clone the parameters and instead of instantiating an error you instantiate a closure